### PR TITLE
#160246870 Bg fix key error on get endpoint

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
**What does this PR do?**
- Fixes bug when a client sends a get request to signup and login endpoint.

**Description of tasks to be completed**
- Define methods on *signup* and *login* endpoints to act on *get* requests from clients.
- Move JWT generation to *backends.py*
- Delete *token.py* from the project since its functions have been moved to *backends.py*

**How should this be tested?**
- Clone this repository
- Checkout to bg-fix-key-error-on-get-endpoint-160246870
- Create and activate a virtual environment for this project if you have done that already
- Install project dependencies by running `pip install -r requirements.txt`
- Create a `.env` file and inside it set `SECRET_KEY` and `DATABASE_URL`
- Run the project from the terminal by issuing `python manage.py runserver`
- using a browser of your choice, navigate to `http://127.0.0.1:8000/api/users/signup/` or `http://127.0.0.1:8000/api/users/login/

**Any background context?**
Providing descriptive responses to users when they use methods or format which are not allowed is good for a better user experience.

**What are the relevant pivotal tracker stories?**
[#160246870](https://www.pivotaltracker.com/story/show/160246870)

**Screenshot**
image

### CODE REVIEW

**Comment by pabishai** [comment]
@meshack-mbuvi Review your PR structure. Look at the "How do I manually test this". It is indented.

**Comment by meshack-mbuvi** [comment]
@PAbishai it's already done.

**Comment by Bileonaire** [comment]
I have checked out the change in the get request for the endpoint in the browser and it is okay.

**Comment by pabishai** [comment]
@meshack-mbuvi It is working as the screenshot displays. However, I do not have a good context of the bug based on the description.

**Comment by mikenthiwa** [comment]
I have reviewed and the bug has been fixed

File: authors/apps/authentication/renderers.py [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+        """
+        If the view throws an error (such as the user can't be authenticated
+        or something similar), `data` will contain an `errors` key. We want
+        the default JSONRenderer to handle rendering errors, so we need to
+        check for this case.
+        """
         errors = data.get('errors', None)
 
 
         if errors is not None:
```
Comment (parseen254) [inline authors/apps/authentication/renderers.py]: @meshack-mbuvi this line is quite repetitive use `if errors` because in python None is considered falsy.

**Review by parseen254 — CHANGES_REQUESTED** [review]

**Review by parseen254 — APPROVED** [review]

**Pull request merged**